test_util/azure.py

```diff
@@ -35,77 +38,126 @@ def validate_hostname_prefix(prefix):
     assert re.match('^[a-z][a-z0-9-]{1,61}[a-z0-9]$', prefix), 'Invalid DNS prefix: {}'.format(prefix)
 
 
+def check_json_object(obj):
+    """ Simple check to fill in the map for automatic parameter casting
+    JSON objects must be represented as dict at this level
+    """
+    assert isinstance(obj, dict), 'Invalid JSON object: {}'.format(obj)
+    return obj
+
+
+def check_array(arr):
+    """ Simple check to fill in the map for automatic parameter casting
+    JSON arrays must be represented as lists at this level
+    """
+    assert isinstance(arr, list), 'Invalid array: {}'.format(arr)
+    return arr
+
+
 class AzureWrapper:
-    def __init__(self, subscription_id: str, client_id: str, client_secret: str, tenant_id: str):
-        self.credentials = azure.common.credentials.ServicePrincipalCredentials(
+    def __init__(self, location: str, subscription_id: str, client_id: str, client_secret: str, tenant_id: str):
+        self.credentials = ServicePrincipalCredentials(
             client_id=client_id,
             secret=client_secret,
             tenant=tenant_id)
         self.rmc = ResourceManagementClient(self.credentials, subscription_id)
         self.nmc = NetworkManagementClient(self.credentials, subscription_id)
+        # location is included to keep a similar model as test_util.aws.BotoWrapper
+        self.location = location
 
-
-class DcosAzureResourceGroup:
-    def __init__(self, group_name, azure_wrapper):
-        self.group_name = group_name
-        self.azure_wrapper = azure_wrapper
-
-    @classmethod
-    def deploy_acs_template(
-            cls, azure_wrapper, template_uri, location, group_name,
-            public_key, master_prefix, agent_prefix, admin_name, oauth_enabled: Union[bool, str],
-            vm_size, agent_count, name_suffix, vm_diagnostics_enabled: bool):
-        """ Creates a new resource group and deploys a ACS DC/OS template to it
-        """
-        assert master_prefix != agent_prefix, 'Master and agents must have unique prefixs'
-        validate_hostname_prefix(master_prefix)
-        validate_hostname_prefix(agent_prefix)
-
+    def deploy_template_to_new_resource_group(self, template_url, group_name, parameters):
         deployment_name = DEPLOYMENT_NAME.format(group_name)
-
         # Resource group must be created before validation can occur
-        if azure_wrapper.rmc.resource_groups.check_existence(group_name):
+        if self.rmc.resource_groups.check_existence(group_name):
             raise Exception("Group name already exists / taken: {}".format(group_name))
         log.info('Starting resource group_creation')
         with contextlib.ExitStack() as stack:
```

Comment:
Yes, finally executes always. We do not want to delete the resource group if everything for deployment is looking green, so then we pop it from the stack before the context exits